### PR TITLE
refactor(encoding): reduce DedupPkCellBasedRowDeserializer initialization overhead

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
  Refactor dedup pk deserializer constructor.
  Instantiation should have low overhead.
  See: https://github.com/singularity-data/risingwave/pull/3578#discussion_r912321346

- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
LGTM. And I believe eventually all the information should be part of `StateTable` or `StateTableDesc`. We should move towards removing the `ColumnDescMapping` thing in the future 🤣

File: src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs [hunk 1]
```diff
@@ -35,36 +35,47 @@ pub struct DedupPkCellBasedRowDeserializer<Desc: Deref<Target = ColumnDescMappin
...
-                if column_desc.data_type.mem_cmp_eq_value_enc() {
-                    column_mapping
-                        .get(column_desc.column_id)
-                        .map(|(_, index)| index)
-                } else {
-                    None
-                }
-            })
-            .collect();
+pub fn create_pk_deserializer_from_pk_descs(
```
Comment (BowenXiao1999) [inline src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs:38]: Add some doc or example on pub function?
Can not get quickly that why this change is necessary.

File: src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs [hunk 2]
```diff
@@ -35,36 +35,47 @@ pub struct DedupPkCellBasedRowDeserializer<Desc: Deref<Target = ColumnDescMappin
...
+            (
+                ordered_desc.column_desc.data_type.clone(),
+                ordered_desc.order,
+            )
+        })
+        .unzip();
+    OrderedRowDeserializer::new(pk_data_types, pk_order_types)
+}
+
+pub fn create_pk_to_row_mapping(
```
Comment (BowenXiao1999) [inline src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs:53]: e.g. How is this pk mapping different from column_mapping?

File: src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs (hunk 1 above)
Comment (kwannoel) [reply src/storage/src/encoding/dedup_pk_cell_based_row_deserializer.rs:45]: Good catch, updated it.

**Review by BowenXiao1999 — APPROVED** [review]

**Review by wcy-fdu — APPROVED** [review]
